embassy-stm32f4/src/exti.rs

```diff
@@ -0,0 +1,128 @@
+//! Async low power Serial.
+//!
+//! The peripheral is autmatically enabled and disabled as required to save power.
+//! Lowest power consumption can only be guaranteed if the send receive futures
+//! are dropped correctly (e.g. not using `mem::forget()`).
+
+use core::future::Future;
+use core::mem;
+use core::pin::Pin;
+
+use embassy::gpio::{WaitForFallingEdge, WaitForRisingEdge};
+use embassy::interrupt::OwnedInterrupt;
+use embassy::util::Signal;
+
+use crate::hal::gpio::{Edge, ExtiPin as HalExtiPin};
+use crate::hal::prelude::*;
+use crate::hal::syscfg::SysCfg;
+use crate::pac::EXTI;
+
+use crate::interrupt;
+
+static mut exti: mem::MaybeUninit<EXTI> = mem::MaybeUninit::<EXTI>::uninit();
+
+
+pub struct ExtiContainer {}
+
+impl ExtiContainer {
+    pub fn new(ex: EXTI) -> Self {
+        unsafe { exti.write(ex); }
+
+        Self{}
+    }
+}
+
+
+/// Interface to the Serial peripheral
+pub struct ExtiPin<T: HalExtiPin, I: OwnedInterrupt> {
```

Comment:
There's nothing that enforces the interrupt is the right one for the pin. User could pass any interrupt and it'd compile fine but not work.

With an `Instance` trait you could have an `Interrupt` associated type, to force user to pass `T::Instance` instead of any interrupt.

Reply:
Was lazy, but will fix it.